packages/neuron-ui/src/components/Send/hooks.ts

```diff
@@ -264,6 +266,52 @@ const clear = (dispatch: StateDispatch) => {
 
 const useClear = (dispatch: StateDispatch) => useCallback(() => clear(dispatch), [dispatch])
 
+export type FeeRateValueArrayItemType = {
+  feeRateValue: string
+  feeValue: string
+}
+
+export const useGetBatchGeneratedTx = async ({
+  walletID,
+  priceArray = [],
+  items,
+}: {
+  walletID: string
+  priceArray?: string[]
+  items: Readonly<State.Output[]>
+}) => {
+  const getUpdateGeneratedTx = (params: Controller.GenerateTransactionParams) =>
+    generateTx(params)
+      .then((res: ControllerResponse<State.Transaction>) => {
+        if (res.status === 1) {
+          return (res as SuccessFromController).result
+        }
+        return res
+      })
+      .catch(() => {
+        return undefined
+      })
+
+  const realParams = {
+    walletID,
+    items: items.map(item => ({
+      address: item.address!,
+      capacity: CKBToShannonFormatter(item.amount, item.unit),
+      date: item.date,
+    })),
+  }
+
+  const requestArray = priceArray.map(itemPrice => getUpdateGeneratedTx({ ...realParams, feeRate: itemPrice }))
+  const allPromiseResult = await Promise.allSettled<State.Transaction>(requestArray)
```

Comment:
`getUpdateGeneratedTx` seems will not reject. And `Promise.allSettled` result will be `{status: 'fulfilled' | 'rejected', value: any, reason: any }`. 
It can return `State.Transaction[]`?